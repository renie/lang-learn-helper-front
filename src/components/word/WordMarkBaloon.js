const WordMarkBaloon = reference => 
    `<div class="baloon" contentEditable="false">
        <p>
            ${reference.name}: a(n) ${reference.meanings[0].wordClass} that means ${reference.meanings[0].description}</p>
        <p>
            As in: ${reference.meanings[0].example}</p>
    </div>`


export default WordMarkBaloon
