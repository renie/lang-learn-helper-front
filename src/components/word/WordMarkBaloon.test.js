import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import WordMarkBaloon from './WordMarkBaloon'

it('renders correctly', () =>{
    const tree = WordMarkBaloon({
            "name":"used",
            "meanings":[{
                "id": 1,
                "wordClass": "verb",
                "example": "He used my pen.",
                "description": "Verb use in past tense."
            },
            {
                "id":2,
                "wordClass":"adjective",
                "example":"My pen is used.",
                "description":"Means the object is not new anymore."
            }],
            "_id":"BcNdVZy038MWxqrg"
        })

    expect(tree).toMatchSnapshot()
})
