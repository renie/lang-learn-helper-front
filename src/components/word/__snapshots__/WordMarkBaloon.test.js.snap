// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<div class=\\"baloon\\" contentEditable=\\"false\\">
        <p>
            used: a(n) verb that means Verb use in past tense.</p>
        <p>
            As in: He used my pen.</p>
    </div>"
`;
